desktop/core/src/desktop/api2_tests.py

```diff
@@ -700,6 +700,27 @@ def test_create(self):
     )
 
 
+  def test_multiple_gist_dirs_on_gist_create(self):
+    home_dir = Directory.objects.get_home_directory(self.user)
+
+    gist_dir1 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+    gist_dir2 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+
+    assert_equal(2, Directory.objects.filter(name=Document2.GIST_DIR, type='directory').count())
+
+    # get_gist_directory merges all duplicate gist directories into one
+    response = self._create_gist(
+      statement='SELECT 12345',
+      doc_type='hive-query',
+      name='test_gist_create',
+    )
+    gist_uuid = json.loads(response.content)['uuid']
+    gist_parent_uuid = Document2.objects.get(uuid=gist_uuid).parent_directory.uuid
+
+    assert_equal(1, Directory.objects.filter(name=Document2.GIST_DIR, type='directory').count())
+    assert_true(Directory.objects.filter(name=Document2.GIST_DIR, type='directory', uuid=gist_parent_uuid).exists())
+
```

Comment:
Last main comment:
could we check here that the Directory is the same as gist_dir1? (as we are supposed to have deleted all but the first one)

Also we don't delete the objects but this is good.